src/ota_metadata/legacy2/metadata.py

```diff
@@ -356,8 +370,31 @@ def iter_dir_entries(self, *, batch_size: int) -> Generator[FileTableDirectories
     def iter_non_regular_entries(
         self, *, batch_size: int
     ) -> Generator[FileTableNonRegularFiles]:
-        with FileTableNonRegularORM(self.connect_fstable()) as pool:
-            yield from pool.orm_select_all_with_pagination(batch_size=batch_size)
+        """Yield entries from base file_table which digest presented in OTA image's file_table.
+
+        This is for assisting faster delta_calculation without full disk scan.
+        """
+        with FileTableNonRegularORM(self.connect_fstable()) as orm:
+            yield from orm.orm_select_all_with_pagination(batch_size=batch_size)
+
+    def iter_common_regular_entries_by_digest(
+        self, base_file_table: StrOrPath, *, max_num_of_entries_per_digest: int = 10
+    ) -> Generator[tuple[bytes, list[FileTableRegularFiles]]]:
+        _hash, _cur = b"", []
+        with FileTableRegularORM(self.connect_fstable()) as orm:
+            for entry in orm.iter_common_by_digest(str(base_file_table)):
+                if entry.digest == _hash:
+                    # When there are too many entries for this digest, just pick the first
+                    #   <max_num_of_entries_per_digest> of them.
+                    if len(_cur) <= max_num_of_entries_per_digest:
+                        _cur.append(entry)
```

Comment:
To avoid the case of there are a lot of entries that have the same hashes, we only get the first 10 entries, this should be enough for most cases.